xds/src/main/java/io/grpc/xds/LookasideLb.java

```diff
@@ -276,6 +425,8 @@ private static ManagedChannel initLbChannel(
 
     @Override
     public void onEndpointChanged(EndpointUpdate endpointUpdate) {
+      channelLogger.log(ChannelLogLevel.DEBUG, "Received an EDS update: '%s'",  endpointUpdate);
```

Comment:
You should not log/expose any thing about xDS protocol here. It's all encapsulated in `XdsClient` (`XdsClient` has such loggings). LBs should have no knowledge about xDS, all they are doing should be based on data (i.e., `ClusterUpdate` and `EndpointUpdate`) given to them and interactions with `XdsClient` interface.

Reply:
I don't see `XdsClient` has such loggings on `EndpointUpdate`.

Reply:
I mean the logging for receive EDS responses. Anyway, my point is LBs should have no idea about xDS protocol and management server. (I shouldn't be expected to see any LDS/RDS/CDS/EDS words in this class.)

Reply:
This EDS policy / EDS balancer, so EDS is inevitable. What about "EDS policy received an update '%s'"?

Reply:
Ok with your new log message. My main points is, I can see from your implementation that you are expecting or assuming a lot things outside this class, such as EDS LB implementations having awareness of how endpoint data is gotten from traffic director. It really should not. The only thing it should aware is endpoint data is gotten from `EndpointWatcher.onEndpointChanged(...)` for watcher added to `XdsClient`. If `XdsClient` does not give you, you have nothing.